lib/java_buildpack/framework/takipi_agent.rb

```diff
@@ -0,0 +1,97 @@
+# Cloud Foundry Java Buildpack
+#
+# Copyright 2013-2017 the original author or authors.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+require 'fileutils'
+require 'java_buildpack/component/versioned_dependency_component'
+require 'java_buildpack/framework'
+require 'java_buildpack/util/qualify_path'
+
+module JavaBuildpack
+  module Framework
+
+    # Encapsulates the functionality for enabling zero-touch OverOps (fka Takipi) support.
+    class TakipiAgent < JavaBuildpack::Component::VersionedDependencyComponent
+      include JavaBuildpack::Util
+
+      # (see JavaBuildpack::Component::BaseComponent#compile)
+      def compile
+        download_tar
+        @droplet.copy_resources
+      end
+
+      # (see JavaBuildpack::Component::BaseComponent#release)
+      def release
+        java_opts = @droplet.java_opts
+        java_opts.add_agentpath(@droplet.sandbox + 'lib/libTakipiAgent.so')
+        application_name java_opts
+        default_env_vars
+        config_env_vars
+      end
+
+      protected
+
+      # (see JavaBuildpack::Component::VersionedDependencyComponent#supports?)
+      def supports?
+        @configuration['secret_key'] || @configuration['collector_host']
```

Comment:
thanks for the detailed explanation. I did see all other agents do that but it kinda breached my cf knowledge so i brute forced my way in using config. I'll try and incorporate the services module and report back if i get stuck